kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -0,0 +1,475 @@
+       * t1 ------------------------ Txn1 starts.
+       * t2 ------- Txn2 starts.
+       * t3 ------- Txn2 commits.
+       * t4 ------- Txn3 starts.
+       * t5 ------- Txn3 commits.
+       * t6 ------------------------ Txn1 commits (SUCCESS).
+       */
+      val dm1 = new DomainMetadata("domain1", "", false)
+      val dm2 = new DomainMetadata("domain2", "", false)
+      val dm3 = new DomainMetadata("domain3", "", false)
+
+      validateDomainMetadataConflictResolution(
+        engine,
+        tablePath,
+        currentTxn1DomainMetadatas = Seq(dm1),
+        winningTxn2DomainMetadatas = Seq(dm2),
+        winningTxn3DomainMetadatas = Seq(dm3),
+        expectedConflict = false
+      )
+    }
+  }
+
+  test(
+    "Conflict resolution - three concurrent txns have DomainMetadata w/ conflicting domains"
+  ) {
+    withTempDirAndEngine { (tablePath, engine) =>
+      /**
+       * Txn1: include DomainMetadata action for "domain1".
+       * Txn2: include DomainMetadata action for "domain2".
+       * Txn3: include DomainMetadata action for "domain1".
+       *
+       * t1 ------------------------ Txn1 starts.
+       * t2 ------- Txn2 starts.
+       * t3 ------- Txn2 commits.
+       * t4 ------- Txn3 starts.
+       * t5 ------- Txn3 commits.
+       * t6 ------------------------ Txn1 commits (FAIL).
+       */
+      val dm1 = new DomainMetadata("domain1", "", false)
+      val dm2 = new DomainMetadata("domain2", "", false)
+      val dm3 = new DomainMetadata("domain1", "", false)
+
+      validateDomainMetadataConflictResolution(
+        engine,
+        tablePath,
+        currentTxn1DomainMetadatas = Seq(dm1),
+        winningTxn2DomainMetadatas = Seq(dm2),
+        winningTxn3DomainMetadatas = Seq(dm3),
+        expectedConflict = true
+      )
+    }
+  }
+
+  test(
+    "Conflict resolution - three concurrent txns have DomainMetadata w/ conflict domains - 2"
+  ) {
+    withTempDirAndEngine { (tablePath, engine) =>
+      /**
+       * Txn1: include DomainMetadata action for "domain1".
+       * Txn2: include DomainMetadata action for "domain1".
+       * Txn3: include DomainMetadata action for "domain2".
+       *
+       * t1 ------------------------ Txn1 starts.
+       * t2 ------- Txn2 starts.
+       * t3 ------- Txn2 commits.
+       * t4 ------- Txn3 starts.
+       * t5 ------- Txn3 commits.
+       * t6 ------------------------ Txn1 commits (FAIL).
+       */
+      val dm1 = new DomainMetadata("domain1", "", false)
+      val dm2 = new DomainMetadata("domain1", "", false)
+      val dm3 = new DomainMetadata("domain2", "", false)
+
+      validateDomainMetadataConflictResolution(
+        engine,
+        tablePath,
+        currentTxn1DomainMetadatas = Seq(dm1),
+        winningTxn2DomainMetadatas = Seq(dm2),
+        winningTxn3DomainMetadatas = Seq(dm3),
+        expectedConflict = true
+      )
+    }
+  }
+
+  test("Integration test - create a table with Spark and read its domain metadata using Kernel") {
```

Comment:
Can we do the reverse as well?

Reply:
Sure, I think this is a good idea! I've added a reverse test (Write with Kernel and read with Spark).